packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx

```diff
@@ -41,22 +43,13 @@ const CountrySelector = ({ onSelect, className }: TCountrySelectorProps) => {
 
     const submitCountry = (values: TValues) => {
         const matching_country = residence_list?.find(c => c.text === values.country_input);
-        if (matching_country) {
-            onSelect(matching_country);
-        }
+        onSelect(matching_country);
     };
 
-    React.useEffect(() => {
-        if (!residence_list) {
-            WS.send();
-        }
-    }, [WS, residence_list]);
-
     return (
         <Formik initialValues={initial_form_values} validate={validateFields} onSubmit={submitCountry}>
             {({ errors, handleBlur, handleChange, setFieldValue, touched, values }) => (
                 <div className={className}>
-                    {!residence_list && is_loading && <Loading is_fullscreen={false} />}
                     {residence_list && residence_list.length && (
```

Comment:
@hamid-deriv What's the point of this conditional rendering if we are not handling other cases? 🤔 
If `residence_list && residence_list.length` is not met then we render blank.

Reply:
When I use WS for fetching the list the select component throws an error when `resident_list` is not initialized. But I think I can remove it now, as the store initial value is `[]`. Thanks for tip :+1:

Reply:
Sorted